### PR TITLE
[RFR][1LP] Automated some Automate tests

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: -v cfme/tests/automate/test_domain.py -k 'test_domain_name_wrong or test_domain_lock_unlock'}}

### CODE REVIEW

File: cfme/automate/explorer/domain.py [hunk 1]
```diff
@@ -304,6 +304,32 @@ def delete(self, cancel=False):
...
                 'Automate Domain "{}": Delete successful'.format(self.description or self.name))
 
+    def lock(self):
+        # Ensure this has correct data
+        self.description
+        details_page = navigate_to(self, 'Details')
+        details_page.configuration.item_select('Lock this Domain')
+        details_page.flash.assert_no_error()
+        details_page.flash.assert_message('The selected Automate Domain were marked as Locked')
+        try:
```
Comment (RonnyPfannschmidt) [inline cfme/automate/explorer/domain.py]: perhaps use `clear_property_cache(self, 'locked')` importable from `utils`

File: cfme/automate/explorer/domain.py [hunk 2]
```diff
@@ -304,6 +304,32 @@ def delete(self, cancel=False):
...
+        assert self.locked
+
+    def unlock(self):
+        # Ensure this has correct data
+        self.description
+        details_page = navigate_to(self, 'Details')
+        details_page.configuration.item_select('Unlock this Domain')
+        details_page.flash.assert_no_error()
+        details_page.flash.assert_message('The selected Automate Domain were marked as Unlocked')
+        try:
```
Comment (RonnyPfannschmidt) [inline cfme/automate/explorer/domain.py]: dito

File: cfme/automate/explorer/domain.py (hunk 1 above)
Comment (mfalesni) [inline cfme/automate/explorer/domain.py]: Good.

File: cfme/tests/automate/test_domain.py [hunk 3]
```diff
@@ -90,3 +90,65 @@ def test_cannot_edit_builtin():
...
+def test_domain_lock_unlock(request):
+    domains = DomainCollection()
+    domain = domains.create(
+        name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha(),
+        enabled=True)
+    request.addfinalizer(domain.delete)
+    ns1 = domain.namespaces.create(name='ns1')
+    ns2 = ns1.namespaces.create(name='ns2')
+    cls = ns2.classes.create(name='class1')
```
Comment (mshriver) [inline cfme/tests/automate/test_domain.py]: **Optional**
`kls` instead of `cls`

File: cfme/tests/automate/test_domain.py [hunk 4]
```diff
@@ -90,3 +90,65 @@ def test_cannot_edit_builtin():
...
+
+
+@pytest.mark.tier(2)
+def test_domain_lock_unlock(request):
+    domains = DomainCollection()
+    domain = domains.create(
+        name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha(),
+        enabled=True)
+    request.addfinalizer(domain.delete)
```
Comment (mshriver) [inline cfme/tests/automate/test_domain.py]: The delete failed against 5.6, Could not find an element u'./ul/li/a[normalize-space(.)="Remove this Domain"]' - might need to version pick the configuration dropdown item.

Its outside your narrow scope here, up to you to fix it here or just write issue.

**Comment by mfalesni** [comment]
@RonnyPfannschmidt Added the fix.

**Pull request merged**